src/elements/fields/QRScanner/index.tsx

```diff
@@ -1,27 +1,42 @@
-import React, { useEffect, useState } from 'react';
-import { FORM_Z_INDEX } from '../../../utils/styles';
+import React, { useCallback, useEffect } from 'react';
 import { dynamicImport } from '../../../integrations/utils';
-import { featheryDoc, featheryWindow } from '../../../utils/browser';
-import { v4 as uuidv4 } from 'uuid';
+import { FORM_Z_INDEX } from '../../../utils/styles';
 
-const QR_SCANNER_URL = 'https://unpkg.com/html5-qrcode';
+import { selectCamera } from './utils/select-camera';
+import { getZoomSettings } from './utils/supports-zoom';
+import { useDeviceRotation } from './hooks/use-device-rotation';
+import { featheryWindow } from '../../../utils/browser';
+import {
+  Html5QrcodeScannerState,
+  PLACEHOLDER_IMAGE,
+  QR_SCANNER_LIB_URL,
+  SCAN_CONFIG
+} from './constants';
 
 let qrPromise = Promise.resolve();
 export function loadQRScanner() {
-  qrPromise = dynamicImport(QR_SCANNER_URL);
+  qrPromise = dynamicImport(QR_SCANNER_LIB_URL);
 }
 
-const onQRError = (qrDivId: string, onChange: any) => () => {
-  const errorEl = featheryDoc().getElementById(`${qrDivId}__header_message`);
-  if (
-    errorEl?.textContent?.trim() ===
-    'D: No MultiFormat Readers were able to detect the code.'
-  ) {
-    errorEl.textContent =
-      'No QR code detected. Please try with a different image.';
+async function createScanner(id: string) {
+  await qrPromise;
+  const window = featheryWindow();
+  for (let i = 0; i < 3; i++) {
```

Comment:
is this for loop relevant anymore given we're no longer using the original library? i feel like we could just await on qrPromise and once it's loaded, we can intiialize the scanner which should be guarnteed to work